### PR TITLE
Some issues

### PR DESCRIPTION
**Pull request by wasade**
Closing out a few more

### CODE REVIEW

**Comment by wasade** [comment]
Should be quick. Lineno commit addresses #284

**Comment by wasade** [comment]
@antgonza @josenavas @ElDeveloper @mortonjt should probably be a quick review as most of these are minor.

File: biom/cli/table_validator.py [hunk 1]
```diff
@@ -493,6 +493,8 @@ def _valid_type(self, table):
         """Table must be a known table type"""
         key = self._json_or_hdf5_key(table, 'type')
         value = self._json_or_hdf5_get(table, key)
+        if value is None or value == "":
+            return "Unknown table table, however that is likely okay."
```
Comment (antgonza) [inline biom/cli/table_validator.py]: table table -> table type?

File: biom/table.py [hunk 2]
```diff
@@ -4280,9 +4324,15 @@ def _extract_data_from_tsv(lines, delim='\t', dtype=float, md_parse=None):
             obs_ids.append(fields[0])
 
             if last_column_is_numeric:
-                values = list(map(dtype, fields[1:]))
+                try:
+                    values = list(map(dtype, fields[1:]))
+                except ValueError:
+                    raise TypeError("Invalid value on line %d." % lineno)
```
Comment (antgonza) [inline biom/table.py]: Should the invalid value also be shown? Here and next block.

File: doc/documentation/biom_format.rst [hunk 3]
```diff
@@ -50,7 +50,7 @@ A sparse representation of an OTU table::
     PC.356 OTU2 7
     PC.356 OTU3 3
 
-OTU table data tends to be sparse (e.g., greater than 90% of counts are zero, and frequently as many as 99% of counts are zero) in which case the latter format is more convenient to work with as it has a smaller memory footprint. Both of these representations are supported in the biom-format project via dense and sparse Table types. Generally if less than 85% of your counts are zero, a dense representation will be more efficient.
+OTU table data tends to be sparse (e.g., greater than 90% of counts are zero, and frequently as many as 99% of counts are zero) in which case the latter format is more convenient to work with as it has a smaller memory footprint. In biom-format 1.0.0, Both of these representations are supported in the biom-format project via dense and sparse Table types. In biom-format 2.x, only sparse is supported as in practice, dense was not useful especially with improved study designs that utilize increasing numbers of samples.
```
Comment (antgonza) [inline doc/documentation/biom_format.rst]: In biom-format 1.0.0, Both -> In biom-format 1.0.0, both

File: doc/documentation/format_versions/biom-2.1.rst [hunk 4]
```diff
@@ -60,7 +60,7 @@ Under the ``observation/group-metadata`` and ``sample/group-metadata`` groups, t
 Example biom files
 ==================
 
-Below is an examples of a rich biom file. To decide which of these you should generate for new data types, see the section on :ref:`sparse-or-dense`.
+Below is an examples of a rich biom file.
```
Comment (antgonza) [inline doc/documentation/format_versions/biom-2.1.rst]: Below is an examples -> Below is an example

**Comment by wasade** [comment]
@antgonza thanks! I think I snagged them all -- and good call on doing the extra work to dump out the bad value as the line number offset was off by 1

**Comment by wasade** [comment]
Thanks! @josenavas @ElDeveloper @mortonjt, available for final review/merge? Bulk is just minor issues.

**Pull request merged**